src/auto-evo/mutations/CommonMutationFunctions.cs

```diff
@@ -0,0 +1,237 @@
+
+                            if (existingOrganelles.CanPlace(result, new List<Hex>(), new List<Hex>()))
+                            {
+                                return result;
+                            }
+                        }
+
+                        // Check every possible rotation value.
+                        for (int rotation = 0; rotation <= 5; ++rotation)
+                        {
+                            result.Orientation = rotation;
+
+                            if (existingOrganelles.CanPlace(result, new List<Hex>(), new List<Hex>()))
+                            {
+                                return result;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        return null;
+    }
+
+    public static void AttachIslandHexes(OrganelleLayout<OrganelleTemplate> organelles)
+    {
+        var workMemory1 = new HashSet<Hex>();
+        var workMemory2 = new List<Hex>();
+        var workMemory3 = new Queue<Hex>();
+        var islandHexes = new List<Hex>();
+
+        organelles.GetIslandHexes(islandHexes, workMemory1, workMemory2, workMemory3);
+
+        // Attach islands
+        while (islandHexes.Count > 0)
+        {
+            var mainHexes = organelles.ComputeHexCache().Except(islandHexes);
+
+            // Compute shortest hex distance
+            Hex minSubHex = default;
+            int minDistance = int.MaxValue;
+            foreach (var mainHex in mainHexes)
+            {
+                foreach (var islandHex in islandHexes)
+                {
+                    var sub = islandHex - mainHex;
+                    int distance = (Math.Abs(sub.Q) + Math.Abs(sub.Q + sub.R) + Math.Abs(sub.R)) / 2;
+                    if (distance < minDistance)
+                    {
+                        minDistance = distance;
+                        minSubHex = sub;
+
+                        // early exit if minDistance == 2 (distance 1 == direct neighbour => not an island)
+                        if (minDistance == 2)
+                            break;
+                    }
+                }
+
+                // early exit if minDistance == 2 (distance 1 == direct neighbour => not an island)
+                if (minDistance == 2)
+                    break;
+            }
+
+            // Calculate the path to move island organelles.
+            // If statement is there because otherwise the path could be (0, 0).
+            if (minSubHex.Q != minSubHex.R)
+                minSubHex.Q = (int)(minSubHex.Q * (minDistance - 1.0) / minDistance);
+
+            minSubHex.R = (int)(minSubHex.R * (minDistance - 1.0) / minDistance);
+
+            // Move all island organelles by minSubHex
+            foreach (var organelle in organelles.Where(o => islandHexes.Any(h =>
+                         o.Definition.GetRotatedHexes(o.Orientation).Contains(h - o.Position))))
```

Comment:
Turning that where into an if statement is pretty easy, but I'm not sure of a good way to handle that Any. With just the if it would still have to capture organelle for the any.

Reply:
Maybe the only option is to just unpack everything as manual loops that build the result of stuff to move?

Reply:
It just took me a second to figure out the best way to go about that. I figured it out now.